mitodl/mitxpro
# chore: upgrade pillow to 9.1.0 & wagtail to 2.15.5

#### Pre-Flight checklist

  - [ ] New settings are documented and present in `app.json` (Not needed in app.json, we won't need to set them through our deployments)

#### What are the relevant tickets?
#2366 

#### What's this PR do?
- Upgrades `Pillow to 9.1.0` (The latest version available as of now)
- Upgrades `wagtail to 2.15.5` (The latest stable +viable version as per Required in order to upgrade Pillow)
- Upgrades `wagtail-metadata to 3.5.0`
- FIxes migration conflicts
- Fixes broken tests for default search backend


#### How should this be manually tested?
Before below, I'd recommend this to be tested 2 ways -- With existing installation & with a fresh installation of xPRO and making sure that things work as expected and there are no new errors in logs.
- Check CI (tests should pass etc)
- Checking things associated with Pillow (This is mostly related to using/manipulating images through wagtail)
- Checking things associated with wagtail (Create new pages along with nested pages and check that things are working as expected) - 
- Specifically, uploading images with both `MITXPRO_USE_S3=True/False`


#### Where should the reviewer start?
Setup wagtail with some content before updating to this PR

## Postflight checklist:
After we have released this PR to RC/Prod we need to do the following
- Since we had to add the default backend now because wagtail deprecated its own default so we need to run `./manage.py wagtail_update_index`
